### ISSUE TITLE
Error thrown in Service Worker when parsing token with multiple dashes or underscores

### ISSUE THREAD

**Issue by sthibert** [AxaFrance/oidc-client#1166]
### Issue and Steps to Reproduce

The function to parse a token in the Service Worker (see below) can throw an error if the "payload" of the token (i.e. `token.split(".")[1]`) contains more than one dash/underscore.

```
function parseJwt(token) {
  return JSON.parse(
    b64DecodeUnicode(token.split(".")[1].replace("-", "+").replace("_", "/"))
  );
}
```

Here, the `replace()` function only replaces the first occurrence of the characters (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace).

---

I unfortunately don't have a "real" test case to demonstrate this error but the following example payload:
```
{
  "sub": "1234567890",
  "name": "John Doe",
  "iat": 1516239022,
  "?": "aa?"
}
```
gives the following token (generated on jwt.io):
```
eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCI_IjoiYWE_In0.5j-qe-UMnFcPUPx-67bQdtZyZYVPoU6m_S-3pu51P9Y
                                                                                                                ^       ^
```
and the following call will throw a DOMException:
```
atob("eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCI_IjoiYWE_In0".replace("-", "+").replace("_", "/"))
```

---

If the error is thrown when decoding the payload of the `id_token`, the variable `_idTokenPayload` will be `null`:
```
_idTokenPayload = extractTokenPayload(tokens.id_token);
```
and a "cannot read properties of null (reading 'nonce')" error will then be thrown in the condition 2 lines below:
```
if (_idTokenPayload.nonce && currentDatabaseElement.nonce != null)
```

### Version

v7.6.0

#### Actual

Only the first occurrence of the characters is replaced and the `atob()` function in `b64DecodeUnicode()` throws an error.

#### Expected

All occurrences of the characters (dash and underscore) should be replaced to properly decode the payload of the token.

Using the `replaceAll()` function instead of `replace()` should fix the issue.
```
function parseJwt(token) {
  return JSON.parse(
    b64DecodeUnicode(token.split(".")[1].replaceAll("-", "+").replaceAll("_", "/"))
  );
}
```

I'd be glad to create a Pull Request but I'm not sure where to apply the fix?

Thank you for your help.

**Comment by sthibert** [comment]
Seeing the browser compatibility of the `replaceAll()` function (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll), it would probably best to use the regex notation in the `replace()` function. Something like `token.split(".")[1].replace(/-/g, "+").replace(/_/g, "/")`.

**Comment by guillaume-chervet** [comment]
hi @sthibert ,

very sorry I missed your issue !
Thank you for all your information to make it more robust !

**Comment by guillaume-chervet** [comment]
Hi @sthibert , I'am totally sorry I completly missed out your really nice issue.

 https://github.com/AxaFrance/oidc-client/pull/1302

**Comment by sthibert** [comment]
Hi @guillaume-chervet, no worries! Actually, the issue has already been fixed in v7.13.16 (https://github.com/AxaFrance/oidc-client/commit/f2310bc917898b779a40edecf3dd73c21ec32b47). 

Note: `replaceAll` is less supported than `replace` so your latest change might trigger other issues on older browsers.
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#browser_compatibility
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#browser_compatibility
